crates/subspace-networking/src/node.rs

```diff
@@ -542,4 +560,46 @@ impl Node {
     pub fn on_connected_peer(&self, callback: HandlerFn<PeerId>) -> HandlerId {
         self.shared.handlers.connected_peer.add(callback)
     }
+
+    /// Returns the request batch handle with common "connection permit" slot from the shared pool.
+    pub async fn get_requests_batch_handle(&self) -> NodeRequestsBatchHandle {
+        let _permit = self.shared.rate_limiter.acquire_regular_permit().await;
```

Comment:
Also I'd never use `_permit` as a variable name if it is actually used later. Use `permit: _permit` because otherwise it is easy to forget to remove when usage of the variable is removed (not the case in this case as the function is trivial, but generally as a rule don't use `_` unless you mean to ignore it).